### PR TITLE
feat: update profile.yml directory logic and nunjucks template

### PR DESCRIPTION
**Pull request by calculuslmvt**
Closes: #10649 and #10674  

### Description:
Lightdash now checks for profile.yml file in parallel logic to how dbt finds profile.yml file using following precedence.
1. `--profiles-dir` option.
2. `DBT_PROFILES_DIR` environment variable.
3. Current working directory using `process.cwd()` which gives directory from which Lightdash process is started. 
4. `~/.dbt/` directory.

note: 
`yarn link` cannot be used to do manually testing of cwd feature as using `yarn lightdash dbt run`  will change the cwd to directory in which package.json is present hence cwd feature will not work. 

### **Bonus Point Issue :** 
`[object object]` error was occurring due to conversion of obejct to string inside `packages/cli/src/dbt/templating.ts`
Now Nunjucks template handles object and insures output is of always of expected type. 

Now `default` keyword can be used to give fallback values in Nunjucks templates. 
Eg: `target: "{{ env_var('DBT_TARGET', default='jaffle') }}"`

Giving no keywords is also supported.
Eg: `target: "{{ env_var('DBT_TARGET', 'jaffle') }}"`

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -30,7 +30,9 @@ const { version: VERSION } = require('../package.json');
 
 const defaultProjectDir = process.env.DBT_PROJECT_DIR || '.';
 const defaultProfilesDir =
-    process.env.DBT_PROFILES_DIR || path.join(os.homedir(), '.dbt');
+    process.env.DBT_PROFILES_DIR ||
+    process.cwd() ||
+    path.join(os.homedir(), '.dbt');
```
[the remark this answers is not in the document]
Comment (calculuslmvt) [reply packages/cli/src/index.ts:35]: Yeah, should have noticed that. 
Happened due to missing testing this case. fixing it.

**Comment by calculuslmvt** [comment]
@ZeRego
Updated logic for `defaultProfilesDir`  

Now, first default directory will be calculated in the logic parallel to dbt, 
then that default profile directory will be used if `--profiles-dir` is not defined. 
Errors for each case will be generated accordingly, as directory calculated by any case will be passed to and checked in `profile.ts` file for validity. 
Created unit tests for all these cases in `profiles.test.ts`
Also created unit test for `default=value` scenario in `templating.test.ts`

File: packages/cli/src/dbt/templating.ts [hunk 2]
```diff
@@ -6,7 +6,15 @@ nunjucksEnv.addFilter('as_number', (str) => parseFloat(str));
 
 // jinja global functions
 const nunjucksContext = {
-    env_var: (key: string, fallback?: string) => process.env[key] ?? fallback,
+    env_var: (key: string, options?: string | { default?: string }) => {
```
[the remark this answers is not in the document]
Comment (calculuslmvt) [reply packages/cli/src/dbt/templating.ts:9]: Done

File: packages/cli/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (calculuslmvt) [reply packages/cli/src/index.ts]: Fixed and created Unit tests.

**Review by ZeRego — APPROVED** [review]
Thanks for the contribution and specially for the unit tests 🏅

**Comment by ZeRego** [comment]
@all-contributors add @calculuslmvt for code

**Pull request merged**